ykrt/pass_finder/cargo_run.py

```diff
@@ -1,22 +1,37 @@
 import os, subprocess, time
 
 def run_test(yk_path,env=None, n=1) :
+    if env is None:
+        env = os.environ.copy()
+    env['PATH'] = f"{yk_path}/bin:{env['PATH']}"
+    env['YK_BUILD_TYPE'] = 'debug'
     prelink_passes = (env or os.environ).get('PRELINK_PASSES', 'Not Set')
     print(f"\033[93mPRELINK_PASSES: {prelink_passes}\033[0m")
-    os.chdir("/home/shreei/research/yk_pv")
-    #subprocess.run(["cargo build"], shell=True, env=env or os.environ)
-    times = []
-    for i in range(n):
-        before = time.time() 
-        c = subprocess.run(["timeout 60 cargo test"], shell=True, env=env or os.environ) 
-        elapsed = time.time() - before
-        times.append(elapsed)
-    mean_time = sum(times) / len(times)
+    os.chdir(yk_path)
+    r = subprocess.run(["timeout 50 cargo test"], shell=True, env=env or os.environ)
+    # assert r.returncode == 0
+    if r.returncode == 0:
+        os.chdir("/home/shreei/research/yklua")
+        times = []
+        for _ in range(n):
+            subprocess.run(["make clean"], shell=True, env=env)
+            c = subprocess.run(["make && sh test.sh"], shell=True, env=env or os.environ)
```

Comment:
What's the difference between `test.sh` and `run.sh`? I guess we also need a `timeout sh test.sh` for the same reasons as above? And probably also `timeout sh run.sh` below?

Reply:
I intend to compile YKLUA and base test cases with test.sh first (this is just to make sure the passes can build and run some YKLUA test cases), `run.sh` is used to run the benchmark test we choose (with Pavel's help) and use that as fitness score metrics.
And thanks I think both of these could use `timeout`.